Sources/Aperture/Aperture.swift

```diff
@@ -1,228 +1,823 @@
-		highlightClicks: Bool = false,
-		screenId: CGDirectDisplayID = .main,
-		audioDevice: AVCaptureDevice? = .default(for: .audio),
-		videoCodec: AVVideoCodecType? = nil,
-		scaleFactor: Double = 1
-	) throws {
-		let input = try AVCaptureScreenInput(displayID: screenId).unwrapOrThrow(Error.invalidScreen)
-
-		input.minFrameDuration = CMTime(videoFramesPerSecond: framesPerSecond)
-
-		if let cropRect = cropRect {
-			input.cropRect = cropRect
-		}
-
-		input.scaleFactor = scaleFactor
-		input.capturesCursor = showCursor
-		input.capturesMouseClicks = highlightClicks
-
-		try self.init(
-			destination: destination,
-			input: input,
-			output: AVCaptureMovieFileOutput(),
-			audioDevice: audioDevice,
-			videoCodec: videoCodec
-		)
-	}
+			if microphoneCaptureSession?.isRunning == true {
+				microphoneCaptureSession?.stopRunning()
+			}
 
-	/**
-	Start a capture session with the given iOS device.
-
-	Use `Aperture.Devices.iOS()` to get a list of connected iOS devices and use the `.id` property to create an `AVCaptureDevice`:
-
-	```
-	AVCaptureDevice(uniqueID: id)
-	```
-
-	The frame rate is 60 frames per second.
-
-	- Parameter destination: The destination URL where the captured video will be saved. Needs to be writable by current user.
-	- Parameter iosDevice: The iOS device to capture.
-	- Parameter audioDevice: An optional audio device to capture.
-	- Parameter videoCodec: The video codec to use when capturing.
-	*/
-	public convenience init(
-		destination: URL,
-		iosDevice: AVCaptureDevice,
-		audioDevice: AVCaptureDevice? = nil,
-		videoCodec: AVVideoCodecType? = nil
-	) throws {
-		let input = try AVCaptureDeviceInput(device: iosDevice)
-
-		try self.init(
-			destination: destination,
-			input: input,
-			output: AVCaptureMovieFileOutput(),
-			audioDevice: audioDevice,
-			videoCodec: videoCodec
-		)
-	}
+			if externalDeviceCaptureSession?.isRunning == true {
+				externalDeviceCaptureSession?.stopRunning()
+			}
 
-	public func start() {
-		session.startRunning()
-		output.startRecording(to: destination, recordingDelegate: self)
-	}
+			if let assetWriter, assetWriter.status == .writing {
+				if let videoInput, assetWriter.inputs.contains([videoInput]) {
+					videoInput.markAsFinished()
+				}
 
-	public func stop() {
-		output.stopRecording()
-	}
+				if let systemAudioInput, assetWriter.inputs.contains([systemAudioInput]) {
+					systemAudioInput.markAsFinished()
+				}
 
-	public func pause() {
-		output.pauseRecording()
-	}
+				if let microphoneInput, assetWriter.inputs.contains([microphoneInput]) {
+					microphoneInput.markAsFinished()
+				}
 
-	public func resume() {
-		output.resumeRecording()
-	}
-}
+				await assetWriter.finishWriting()
+			}
 
-extension Aperture: AVCaptureFileOutputRecordingDelegate {
-	private var shouldPreventSleep: Bool {
-		get { activity != nil }
-		set {
-			if newValue {
-				activity = ProcessInfo.processInfo.beginActivity(options: .idleSystemSleepDisabled, reason: "Recording screen")
```

Comment:
The activity handling could be simplified with something like this:

```swift
final class Activity {
	private let activity: NSObjectProtocol

	init(
		_ options: ProcessInfo.ActivityOptions = [],
		reason: String
	) {
		self.activity = ProcessInfo.processInfo.beginActivity(options: options, reason: reason)
	}

	deinit {
		ProcessInfo.processInfo.endActivity(activity)
	}
}
```

Reply:
The benefit of the class is that it's self-contained you can just `nil` the property in the recorder to end the activity.

Reply:
Sounds good. Updated.

I think all the comments are addressed, I'm going to start working on the readme updates